elasticapm/traces.py

```diff
@@ -717,6 +717,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 logger.debug("ended non-existing span %s of type %s", self.name, self.type)
 
 
+class transaction(object):
+    def __init__(self, name=None, transaction_type=None, trace_parent=None):
+        self.name = name
+        self.transaction_type = transaction_type or "code"
+        self.trace_parent = trace_parent
+
+    def __call__(self, func):
+        self.name = self.name or get_name_from_func(func)
+
+        @functools.wraps(func)
```

Comment:
Is it possible to use `wrapt` here?

Reply:
Btw did it makes sense to PR on this PR? As this implementation lacks some useful cases.

Reply:
@kalemas You're welcome to make a PR, though this PR is pretty out of date so it might make more sense to wait until @beniwohli gets back to it.